lib/base.ts

```diff
@@ -36,6 +36,13 @@ export class TranspilerBase {
     }
   }
 
+  assert(c: ts.Node, condition: boolean, reason: string): void {
+    if (!condition) {
+      this.reportError(c, reason);
+      throw new Error(reason);
```

Comment:
actually, isn't this just `reportError` now? Maybe just `reportError(...); return;` instead of assert?